### DISCUSSION TITLE
Pass Alpine instance to alpine:* events

### DISCUSSION THREAD

**Discussion by septatrix** [alpinejs/alpine#3407] [3. Feature ideas]
By passing the Alpine instance as part of the event details, the instance must no longer be globally accessible and the `window` access can be eliminated. This can be useful for ES modules for keeping everything encapsulated properly while still being able to use Alpine.data etc.

**Comment by ekwoka** [comment]
If you're using modules, you should be using Alpine.plugin, not the event listeners.
